homeassistant/components/emoncms/sensor.py

```diff
@@ -84,19 +90,29 @@ async def async_setup_platform(
     exclude_feeds = config.get(CONF_EXCLUDE_FEEDID)
     include_only_feeds = config.get(CONF_ONLY_INCLUDE_FEEDID)
     sensor_names = config.get(CONF_SENSOR_NAMES)
+    scan_interval = config.get(CONF_SCAN_INTERVAL, timedelta(seconds=30))
 
     if value_template is not None:
         value_template.hass = hass
 
     emoncms_client = EmoncmsClient(url, apikey, session=async_get_clientsession(hass))
-    elems = await emoncms_client.async_list_feeds()
-
+    coordinator: DataUpdateCoordinator[list[dict[str, Any]] | None] = (
```

Comment:
Let's create a new module `coordinator.py` where we create a full blown coordinator instead

Reply:
I thought it more concise this way
In found other integrations using a coordinator and without the coordinator.py 
cf the modbus one :
https://github.com/home-assistant/core/blob/1eb8b5a27c9f1c23c4ce8496c58c2b80394f2512/homeassistant/components/modbus/sensor.py#L91

Reply:
We like to promote the consistency of placement of common patterns in their own module, and thus I think it would be nicer to have it in the `coordinator.py`

Reply:
Creation of a new module `coordinator.py` done